version: "3.1"

services:
  django-socio-grpc:
    image: django-socio-grpc
    tty: true
    build:
      context: .
    environment:
      - DB_NAME=test
      - DB_USER=test
      - DB_PASSWORD=test_pw
      - DB_HOST=test-database
      - DB_PORT=5432
    command: python test_utils/runserver.py
    #command: python test_utils/load_tests.py
    # command: /bin/bash
    links:
      - test-database
    volumes:
      - .:/opt/code
      - ./test_utils:/opt/code/test_utils
    # ports:
      # - "7000:80"
    networks:
      django-socio-grpc-network:

  django-socio-grpc-docs:
    image: django-socio-grpc
    build:
      context: .
    command: mkdocs serve -a 0.0.0.0:6001 -t readthedocs
    volumes:
      - .:/opt/code
    ports:
      - "6001:6001"
    networks:
      django-socio-grpc-network:

  test-database:
    image: library/postgres:15.3
    ports:
      - "5432:5432"
    volumes:
      - django-socio-grpc-data:/var/lib/postgresql/data
    environment:
      POSTGRES_DB: test
      POSTGRES_USER: test
      POSTGRES_PASSWORD: test_pw
    networks:
      django-socio-grpc-network:

networks:
  django-socio-grpc-network:

volumes:
  django-socio-grpc-data: {}
